### DISCUSSION TITLE
uv__count_bufs crash

### DISCUSSION THREAD

**Discussion by JevonSmart** [libuv/libuv#4440] [Q&A]
Hello, my project is using the libuv library, and the program crashes after running in the libuv's uv__count_bufs function. The crash stack trace is as follows. I have been investigating for a long time and have not found the problem. Can anyone help me？

SIGSEGV SEGV_MAPERR
uv__count_bufs(uv-common.c:628)
uv__write_callbacks(stream.c:918)
uv__stream_io(stream.c:1230)
uv__io_poll(kqueue.c:0)
uv_run(core.c:414)
LoopImpl::Run()(message.cc:0)
threading::ThreadImpl::UvCallback()(theread.cc:53)
libsystem_pthread.dylib _pthread_start + 99
libsystem_pthread.dylib thread_start + 15

**Comment by saghul** [comment]
Can you share the code you are running which makes it crash? Are you passing stack allocated buffers to `uv_write` by any chance?

**Reply by JevonSmart** [reply]
hello,
The code related to uv_write is as follows:

```
using buf_byter = std::vector<uint8_t>;
static void on_uv_write_call_back(uv_write_t* req, int status) {
  // xxxx
  
  {
    auto buf = (std::shared_ptr<buf_byter>*)req->data;
    delete buf;
  }

  free(req);
}

bool SendUV(std::shared_ptr<buf_byter> data) {
  uv_buf_t write_buffer;
  write_buffer.base = (char*)data->data();
  write_buffer.len = static_cast<unsigned long>(data->size());
  auto uv_tcp_write = reinterpret_cast<uv_write_t*>(malloc(sizeof(uv_write_t)));
  uv_tcp_write->data = new std::shared_ptr<buf_byter>(std::move(data));
  auto error = uv_write(uv_tcp_write, reinterpret_cast<uv_stream_t*>(uv_tcp_handle_),
    reinterpret_cast<const uv_buf_t*>(&write_buffer), 1, on_uv_write_call_back);
  if (error) {
    free(uv_tcp_write);
    return false;
  }
  return true;
}
```
